eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/core/FluentBuilder.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.core;
+
+import org.dmfs.optional.Optional;
+
+
+/**
+ * Interface for builders that use {@link Key} and {@link Box} to add parameters to their built result object.
+ *
+ * @author Gabor Keszthelyi
+ */
+public interface FluentBuilder<R>
+{
+    R build();
+
+    <T> FluentBuilder<R> withParam(Key<T> key, Box<T> box);
+
+    <T> FluentBuilder<R> withParam(Key<T> key, Optional<Box<T>> optBox);
+}
```

Comment:
I still think that having a method like
```
  <T extends Boxable> FluentBuilder<R> withParam(Key<T> key, Optional<T> box); 
```
could make things easier. You could give it another name to avoid signature conflicts.